scripts/storage-emulator-integration/tests.ts

```diff
@@ -756,6 +759,46 @@ describe("Storage emulator", () => {
       expect(uploadState).to.equal("success");
     });
 
+    it("should should set metadata correctly on multipart uploads", async () => {
+      const customMetadata = {
+        contentDisposition: "initialCommit",
+        contentType: "image/jpg",
+        name: "test_upload.jpg",
+      };
+
+      const uploadURL = await supertest(STORAGE_EMULATOR_HOST)
+        .post(`/v0/b/${storageBucket}/o/test_upload.jpg?uploadType=resumable&name=test_upload.jpg`)
+        .send(customMetadata)
+        .set({
+          Authorization: "Bearer owner",
+          "X-Goog-Upload-Protocol": "resumable",
+          "X-Goog-Upload-Command": "start",
+        })
+        .expect(200)
+        .then((res) => new URL(res.header["x-goog-upload-url"]));
+
+      await supertest(STORAGE_EMULATOR_HOST)
+        .put(uploadURL.pathname + uploadURL.search)
+        .set({
+          "X-Goog-Upload-Protocol": "resumable",
+          "X-Goog-Upload-Command": "upload",
+        })
+        .expect(200);
```

Comment:
do you need this request?

Reply:
Not needed, I added it when I was testing a few things but I'll remove it.